lib/datasource/metadata.ts

```diff
@@ -158,7 +158,7 @@ export function addMetaData(
       let massagedUrl;
       if (parsedUrl.hostname.includes('gitlab')) {
         massagedUrl = massageGitlabUrl(dep.sourceUrl);
-      } else {
+      } else if (!dep.sourceUrl.includes('/tree/')) {
```

Comment:
What does this change do?

Reply:
By default the sourceUrl is cut down to the repo URL (e.g. `https://github.com/emedvedev/renovate`), but since we use subdirs as the source, we need the full URL to the subdir (e.g. `https://github.com/emedvedev/renovate/tree/master/bin`) maintained. The change prevents cutting those URLs